docs/codebase.md

```diff
@@ -52,32 +52,75 @@ File Descriptions
   * Renders the line displayed while resizing a column
   * Used by FixedDataTable.js
 * FixedDataTableColumnReorderHandle.js
-  * Renders the element to reorder a column
+  * Renders the knob used to reorder a column
   * Used by FixedDataTableCell.js
 
-### State Management Helpers
-* FixedDataTableRowBuffer.js
-  * Provides the windowing logic for which rows to display on screen
-  * Used by FixedDataTableBufferedRows.js
-* FixedDataTableScrollHelper.js
-  * Handles the logic for programatically scrolling to a rows
-  * Handles scroll events
-  * Used by FixedDataTable.js
+### State Management  
+* scrollAnchor.js
+  * Finds the vertical scroll 'anchor'
+  * Scroll anchor decides the first/last row's index amd offset from edge of table
+   
+* computeRenderedRows.js
+  * uses the scroll anchor to find the range of rows displayed in the viewport
+  * it also updates the height of these rows using `updateRowHeight`
+
+* updateRowHeight.js
+  * updates the row height, caches it, and keeps total scroll height in sync
+  
+* columnStateHelper.js
+  * finds horizontal scroll offset
+  * also manages column reordering and resizing
+ 
+### Selectors
+* ariaAttributes.js
+  * Calculates the aria roles and attributes to be given to our columns, rows, cells, .etc
+
+* columnTemplates.js
+  * finds the props and template (renderer) for the cells, and categorizes them into a useful format
+  * has lot of logic shared with convertColumnElementsToData and relies on columnWidths
+  * really needs a refactor/cleanup/elimination
+  
+* columnWidths.js
+  * finds the width of each column (and column group if present)
+  * also calculates flex widths
+  * determines max horizontal scroll
+  
+* roughHeights.js
+  * calculates estimates for the dimensions of the content view port
+  * accounts for scroll bar presence and hence gives min/max dimensions
+  * also decides count of buffered rows
+  
+* scrollbarsVisible.js
+  * calculates scrollbar state (i.e., if it should be visible or not needed)
+  * also calculates the height available for the viewport
+
+* tableHeights.js
+  * calcalates height and offset for different components of the table (full table, viewport, scrollbar, etc.)  
+
+### Helpers
 * ReactTouchHandler.js
   * Handles touch events and converts them to scroll events
   * Used by FixedDataTable.js
   * Semi-experimental
+  
 * FixedDataTableTranslateDOMPosition.js
   * Helper to shim Facebook's translateDOMPositionXY to work for server-side rendering
-* FixedDataTableWidthHelper.js
-  * Helper logic for supporting flex column widths
-  * Used by FixedDataTable.js
+  
 * FixedDataTableHelper.js
-  * Dead code - let's delete it!
+  * Dead code - let's delete it! (how dead is this? it's used by FDTCell and FDTCellGroup)
```

Comment:
passing from the root component to `FixedDataTableCell` might be a bit redundant. 

Also since it doesn't rely on any state/props, how about we move it from `src/FixedDataTableHelper.js` to somewhere like `src/helper/directionHelper.js`, and just use the helper directly?